### PR TITLE
Automatically generated .vscode/settings.json file

### PR DESCRIPTION
**Pull request by edwardalee**
This PR generates a .vscode/settings.json file in the src-gen directory so that when you open VSCode in that directory, it can compile the generated program. (Recommend selecting Clang 11 kit when prompted on opening code) .  The CMake debugger also will work within code.

### CODE REVIEW

**Review by erlingrj — COMMENTED** [review]
Great, lets merge!

**Review by erlingrj — APPROVED** [review]

File: org.lflang/src/org/lflang/generator/c/CGenerator.java [hunk 1]
```diff
@@ -547,6 +548,29 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
...
+            if (!Files.exists(vscodePath))
+            Files.createDirectory(vscodePath);
+            FileUtil.writeToFile(
+                settings,
+                Path.of(fileConfig.getSrcGenPath()
+                    + File.separator + ".vscode"
+                    + File.separator + "settings.json")
+            );
+        } catch (IOException e) {
+            Exceptions.sneakyThrow(e);
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CGenerator.java:571]: The `sneakyThrow` pattern was inherited from generated Xtend code, and AFAIK there is no good reason to use it. If we can use the error reporter, that would be best, but just an ordinary `RuntimeException` would also do, I guess.

**Review by lhstrh — APPROVED** [review]
:rocket:

File: org.lflang/src/org/lflang/generator/c/CGenerator.java (hunk 1 above)
Comment (lhstrh) [reply org.lflang/src/org/lflang/generator/c/CGenerator.java:571]: Create #1761.

**Pull request merged**